app/controllers/application_controller.rb

```diff
@@ -269,4 +269,19 @@ def domain_redirection_required?
   def redirect_to_primary_domain
     observable_redirect_to "#{request.ssl? ? 'https' : 'http'}://#{current_school.domains.primary.fqdn}#{request.path}"
   end
+
+  before_action :set_last_seen_at,
+                if:
+                  proc {
+                    user_signed_in? &&
+                      (
+                        session[:last_seen_at] == nil ||
+                          session[:last_seen_at] < 15.minutes.ago
+                      )
+                  }
+
+  def set_last_seen_at
+    current_user.update(last_seen_at: Time.current)
```

Comment:
Prefer the bang version `update!` so that it raises in case of unsuccessful update.